src/cluster_legacy.h

```diff
@@ -337,9 +343,7 @@ struct clusterState {
     dict *nodes;            /* Hash table of name -> clusterNode structures */
     dict *shards;           /* Hash table of shard_id -> list (of nodes) structures */
     dict *nodes_black_list; /* Nodes we don't re-add for a few seconds. */
-    clusterNode *migrating_slots_to[CLUSTER_SLOTS];
-    clusterNode *importing_slots_from[CLUSTER_SLOTS];
-    clusterNode *slots[CLUSTER_SLOTS];
+    clusterSlotState slot_states[CLUSTER_SLOTS];
```

Comment:
When we look up the slot owner, importing from and migrating to for the same slot, we can access only one cache line instead of three. This is what we do in getNodeByQuery.

I can't see any performance improvement on your benchmark results. My guess is `valkey-benchmark` is running on 100% CPU but it can't keep Valkey busy enough to push the throughput higher.

You should maybe run valkey-benchmark with `--threads 4` or something to generate enough traffic to keep valkey busy enough.